src/scaffold/store.ts

```diff
@@ -75,12 +92,34 @@ export const useStore = create<Store>((set) => {
           type: repoType,
         });
       },
+      updateTemplateOptions() {
+        dispatch(set, (state) => ({
+          ...state,
+          templateOptions: ['spinner'],
+        }));
+        const packageName = useStore.getState().state.options.template;
+        vscode.postMessage({
+          command: 'getTemplatesOptions',
+          packageName,
+        });
+      },
       selectLocalPath() {
         vscode.postMessage({
           command: 'getLocalPath',
         });
       },
       updateOptions(newOptions) {
+        if (newOptions.dynamicOptionValues) {
+          dispatch(set, (state) => ({
+            ...state,
+            options: {
+              ...state.options,
+              dynamicOptionValues: [...state.options.dynamicOptionValues, newOptions.dynamicOptionValues],
+            },
+          }));
+          return;
```

Comment:
Instead of `return` use an `else` block below. It should either be this one or the other.